### PR TITLE
Adding the Windows named pipe example, this fixes #20617

### LINKED ISSUE #20617: listen(tempname()) fails on Windows
Linux:
```julia
julia> listen(tempname())
Base.PipeServer(active)

julia> 
```
Windows:
```julia
julia> listen(tempname())
ERROR: ArgumentError: could not listen on path \AppData\Local\Temp\jl_ADD1.tmp
 in listen(::String) at .\stream.jl:944

julia>
```

Both with version v"0.5.0"

Documentation propose this should work also in Windows: http://docs.julialang.org/en/stable/manual/networking-and-streams/#a-simple-tcp-example

> ```julia
> julia> listen("testsocket") # Listens on a UNIX domain socket/named pipe
> PipeServer(active)
> ```

> Note that the return type of the last invocation is different. This is because this server does not listen on TCP, but rather on a named pipe (Windows) or UNIX domain socket.

### PR DESCRIPTION
**Pull request by TeroFrondelius**
This is a small clarification to the document. See https://github.com/JuliaLang/julia/issues/20617

### CODE REVIEW

**Review by vtjnash — APPROVED** [review]

**Comment by musm** [comment]
merge?

File: doc/src/manual/networking-and-streams.md [hunk 1]
```diff
@@ -216,15 +216,21 @@ TCPServer(active)
...
 ```
 
 Note that the return type of the last invocation is different. This is because this server does
-not listen on TCP, but rather on a named pipe (Windows) or UNIX domain socket. The difference
-is subtle and has to do with the [`accept()`](@ref) and [`connect()`](@ref) methods. The [`accept()`](@ref)
-method retrieves a connection to the client that is connecting on the server we just created,
-while the [`connect()`](@ref) function connects to a server using the specified method. The [`connect()`](@ref)
+not listen on TCP, but rather on a named pipe (Windows) or UNIX domain socket.
+Also note that Windows named pipe format has to be a specific pattern such that
+their name prefix (\\.\pipe\) uniquely identifies the [filetype](https://msdn.microsoft.com/en-us/library/windows/desktop/aa365783(v=vs.85).aspx).
```
Comment (tkelman) [inline doc/src/manual/networking-and-streams.md]: `\\.\pipe\` should be code quoted

**Comment by musm** [comment]
Also should we consider comment https://github.com/JuliaLang/julia/issues/20617#issuecomment-280086480
for a cross platform version? 

> How about adding a new function: tempsocket() or tempnamedpipe()? Which would return correctly named pipe with random temporyname for Windows and tempname() for Linux.

**Comment by tkelman** [comment]
included in #22695

**Pull request closed without merging**